src/FsAutoComplete.Core/Commands.fs

```diff
@@ -944,6 +924,98 @@ module Commands =
     }
 
 
+  /// * `includeDeclarations`:
+  ///   if `false` only returns usage locations and excludes declarations
+  ///   * Note: if `true` you can still separate usages and declarations from each other
+  ///     with `Symbol.partitionInfoDeclarationsAndUsages`
+  /// * `includeBackticks`:
+  ///   if `true` returns ranges including existing backticks, otherwise without:
+  ///   `let _ = ``my value`` + 42`
+  ///   * `true`: ` ``my value`` `
+  ///   * `false`: `my value`
+  /// * `errorOnFailureToFixRange`:
+  ///   Ranges returned by FCS don't just span the actual identifier, but include Namespace, Module, Type: `System.String.IsNullOrEmpty`
+  ///   These ranges gets adjusted to include just the concrete identifier (`IsNullOrEmpty`)
+  ///   * If `false` and range cannot be adjust, the original range gets used.
+  ///     * When results are more important than always exact range
+  ///       -> for "Find All References"
+  ///   * If `true`: Instead of using the source range, this function instead returns an Error
+  ///     * When exact ranges are required
+  ///       -> for "Rename"
+  let symbolUseWorkspace
+    (getDeclarationLocation: FSharpSymbolUse * NamedText -> Async<SymbolDeclarationLocation option>)
+    (findReferencesForSymbolInFile: (string<LocalPath> * FSharpProjectOptions * FSharpSymbol) -> Async<Range seq>)
+    (tryGetFileSource: string<LocalPath> -> Async<ResultOrString<NamedText>>)
+    (tryGetProjectOptionsForFsproj: string<LocalPath> -> Async<FSharpProjectOptions option>)
+    (getAllProjectOptions: unit -> Async<FSharpProjectOptions seq>)
+    (includeDeclarations: bool)
+    (includeBackticks: bool)
+    (errorOnFailureToFixRange: bool)
+    pos
+    lineStr
+    (text: NamedText)
+    (tyRes: ParseAndCheckResults)
+    : Async<Result<(FSharpSymbol * IDictionary<string<LocalPath>, Range[]>), string>> =
+    asyncResult {
+      let! symbolUse = tyRes.TryGetSymbolUse pos lineStr |> Result.ofOption (fun _ -> "No symbol")
+
+      return!
+        symbolUseWorkspaceAux
+          getDeclarationLocation
+          findReferencesForSymbolInFile
+          tryGetFileSource
+          tryGetProjectOptionsForFsproj
+          getAllProjectOptions
+          includeDeclarations
+          includeBackticks
+          errorOnFailureToFixRange
+          text
+          tyRes
+          symbolUse
+    }
+
+  let symbolUseWorkspace2
```

Comment:
Should we keep a separate implementation or could you leave a comment for why there's a second implementation?

Reply:
Ah yes, we still mean to clean that up. Not sure if we should rename or consider moving everything to the v2 implementation.

Reply:
I made a version that uses v2 for everything and takes the simplest approach 😏 
https://github.com/nojaf/FsAutoComplete/pull/5
@TheAngryByrd  Let me know, if you prefer a version that keeps the symbol(s) in it's return type.